pkg/controller/util.go

```diff
@@ -546,28 +554,74 @@ func podSucceededFromPVC(pvc *v1.PersistentVolumeClaim) bool {
 	return (podPhaseFromPVC(pvc) == v1.PodSucceeded)
 }
 
-func setConditionFromPodWithPrefix(anno map[string]string, prefix string, pod *v1.Pod) {
-	if pod.Status.ContainerStatuses != nil {
-		if pod.Status.ContainerStatuses[0].State.Running != nil {
-			anno[prefix] = "true"
-			anno[prefix+".message"] = ""
-			anno[prefix+".reason"] = podRunningReason
-		} else {
-			anno[AnnRunningCondition] = "false"
-			if pod.Status.ContainerStatuses[0].State.Waiting != nil {
-				anno[prefix+".message"] = pod.Status.ContainerStatuses[0].State.Waiting.Message
-				anno[prefix+".reason"] = pod.Status.ContainerStatuses[0].State.Waiting.Reason
-			} else if pod.Status.ContainerStatuses[0].State.Terminated != nil {
-				anno[prefix+".message"] = pod.Status.ContainerStatuses[0].State.Terminated.Message
-				anno[prefix+".reason"] = pod.Status.ContainerStatuses[0].State.Terminated.Reason
-				if strings.Contains(pod.Status.ContainerStatuses[0].State.Terminated.Message, common.PreallocationApplied) {
-					anno[AnnPreallocationApplied] = "true"
-				}
+func setAnnotationsFromPodWithPrefix(anno map[string]string, prefix string, pod *v1.Pod) {
```

Comment:
Maybe a nitpick: With such a name It would be easier to read if the parameter names follow what is mentioned in the name so:
the name states setAnnotationsFromPodWithPrefix(anno, pod, prefix)

Reply:
Done